src/stores/BlockchainStore.ts

```diff
@@ -166,33 +166,43 @@ export default class BlockchainStore {
         } = this.context;
         const networkName = configStore.getActiveChainName();
 
-        if (!daoStore.getCache()) {
+        const cache = await caches.open(`dxvote-cache`);
+        let match = await cache.match(networkName);
+        let networkCache = null;
+        if (match) {
+          networkCache = JSON.parse(await match.text());
+        }
+
+        const blockNumber = (await library.eth.getBlockNumber()) - 1;
+
+        // Fetch cache from ipfs if not in localStorage or > ~4 days old

```

Comment:
Or if the hash of the cache saved in storage is different to the one hosted online.You will have to also store the IPFS hash of the cache stored.

Bacause if you get the cache now with hash A and it changes today to hash B it will only be fetched 3 days from today, right? So first lets check the hash of the cache stored and then time.

Reply:
Oh smart! So I can remove the 4 days check, store the hash and check it agains the most recent one for whether it needs fetched or not